### PR TITLE
Renamed `HardwareFamily` to `DeviceFamily`.

### LINKED ISSUE #5352: Rename HardwareFamily to DeviceFamily
In light of some recent future modeling conversations, we will rename the new HardwareFamily model to DeviceFamily. No other changes to the model.

### PR DESCRIPTION
**Pull request by HanlinMiao**
# Closes #5352 
# What's Changed

Pretty Straightforward change here but please do a sanity check to see if I missed any places.

# Screenshots

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: changes/3559.added [hunk 1]
```diff
@@ -1,2 +1,2 @@
-Added `HardwareFamily` model class.
-Added `hardware_family` field to Device Type model class.
+Added `DeviceFamily` model class.
```
Comment (timizuoebideri1) [inline changes/3559.added:1]: I think these are changes for issue #3559, and should be left that way?

File: nautobot/dcim/filters/__init__.py [hunk 2]
```diff
@@ -528,8 +528,8 @@ class DeviceTypeFilterSet(NautobotFilterSet):
     manufacturer = NaturalKeyOrPKMultipleChoiceFilter(
         queryset=Manufacturer.objects.all(), to_field_name="name", label="Manufacturer (name or ID)"
     )
-    hardware_family = NaturalKeyOrPKMultipleChoiceFilter(
-        queryset=HardwareFamily.objects.all(), to_field_name="name", label="Hardware family (name or ID)"
+    device_family = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=DeviceFamily.objects.all(), to_field_name="name", label="Hardware family (name or ID)"
```
Comment (timizuoebideri1) [inline nautobot/dcim/filters/__init__.py:532]:
```suggestion
        queryset=DeviceFamily.objects.all(), to_field_name="name", label="Device family (name or ID)"
```

File: nautobot/dcim/filters/__init__.py [hunk 3]
```diff
@@ -848,9 +848,9 @@ class DeviceFilterSet(
...
         label="Manufacturer (name or ID)",
     )
-    hardware_family = NaturalKeyOrPKMultipleChoiceFilter(
-        field_name="device_type__hardware_family",
-        queryset=HardwareFamily.objects.all(),
+    device_family = NaturalKeyOrPKMultipleChoiceFilter(
+        field_name="device_type__device_family",
+        queryset=DeviceFamily.objects.all(),
         to_field_name="name",
         label="Hardware family (name or ID)",
```
Comment (timizuoebideri1) [inline nautobot/dcim/filters/__init__.py:855]:
```suggestion
        label="Device family (name or ID)",
```

File: changes/3559.added (hunk 1 above)
Comment (glennmatthews) [reply changes/3559.added:1]: We don't have a defined standard for maintaining "pre-release" changelog fragments, but I understand the reasoning here - it's potentially confusing to the user when we ship 2.2 for the "added" to refer to things that weren't added as described. On the flip side, it's good for the change fragments to document the actual work done. Perhaps something like the following might be a good compromise?

```suggestion
Added `HardwareFamily` model class. (Renamed before release to `DeviceFamily`.)
```

File: nautobot/dcim/api/urls.py [hunk 4]
```diff
@@ -15,7 +15,7 @@
 router.register("rack-reservations", views.RackReservationViewSet)
 
 # Device types
-router.register("hardware-families", views.HardwareFamilyViewSet)
+router.register("device-families", views.DeviceFamilyViewSet)
 router.register("manufacturers", views.ManufacturerViewSet)
```
Comment (glennmatthews) [inline nautobot/dcim/api/urls.py:19]: Nit, but maybe swap the order of these two for consistency?

File: nautobot/dcim/models/devices.py [hunk 5]
```diff
@@ -83,17 +83,17 @@ def __str__(self):
...
+    A Device Family is a model that represents a grouping of DeviceTypes.
     """
 
     name = models.CharField(max_length=CHARFIELD_MAX_LENGTH, unique=True)
     description = models.CharField(max_length=CHARFIELD_MAX_LENGTH, blank=True)
 
     class Meta:
         ordering = ["name"]
-        verbose_name_plural = "hardware families"
+        verbose_name_plural = "Device Families"
```
Comment (glennmatthews) [inline nautobot/dcim/models/devices.py:96]:
```suggestion
        verbose_name_plural = "device families"
```

File: nautobot/dcim/migrations/0053_create_device_family_model.py [hunk 6]
```diff
@@ -38,7 +38,7 @@ class Migration(migrations.Migration):
             ],
             options={
                 "ordering": ["name"],
-                "verbose_name_plural": "hardware families",
+                "verbose_name_plural": "Device Families",
```
Comment (glennmatthews) [inline nautobot/dcim/migrations/0053_create_device_family_model.py:41]:
```suggestion
                "verbose_name_plural": "device families",
```

File: nautobot/dcim/tables/__init__.py [hunk 7]
```diff
@@ -80,7 +80,7 @@
     "DeviceTypeTable",
     "FrontPortTable",
     "FrontPortTemplateTable",
-    "HardwareFamilyTable",
+    "DeviceFamilyTable",
```
Comment (glennmatthews) [inline nautobot/dcim/tables/__init__.py:83]: fix ordering here please

File: nautobot/dcim/navigation.py [hunk 8]
```diff
@@ -227,17 +227,17 @@
                         ),
                     ),
                     NavMenuItem(
```
Comment (glennmatthews) [inline nautobot/dcim/navigation.py:229]: Were we going to move `Device Families` above `Manufacturers` in the nav as a part of this PR, or is that still up in the air? @lampwins

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**